modules/core/keeper/msg_server_test.go

```diff
@@ -1530,6 +1530,71 @@ func (suite *KeeperTestSuite) TestChannelUpgradeCancel() {
 				suite.Require().Equal(uint64(2), channel.UpgradeSequence)
 			},
 		},
+		{
+			"success: keeper is authority & channel state in FLUSHING, so error receipt is ignored and channel is restored to initial upgrade sequence",
```

Comment:
love all the test cases! I think one more with authority and channel in FLUSHCOMPLETE would be great. Showing that a proof is required